app/models/estimated_population.rb

```diff
@@ -35,13 +22,18 @@ def hypertension_patient_coverage_rate
     return rate if rate > 0.0
   end
 
-  def show_coverage
+  memoize def show_coverage
     if region.district_region?
       !!region.estimated_population&.hypertension_patient_coverage_rate
     elsif region.state_region?
-      region.estimated_population&.is_population_available_for_all_districts
+      region.estimated_population&.population_available_for_all_districts?
     else
       false
     end
   end
+
+  def population_available_for_all_districts?
+    state = region.state_region
+    state.district_regions.includes(:estimated_population).all? { |d| d.estimated_population }
```

Comment:
This will load all the related population records in one query instead of one query for every district.

Reply:
Much simpler :) This is actually where I found the bug. I appended `.population` to `d.estimated_population` to check if the `population` field is set.